include/zephyr/bluetooth/bluetooth.h

```diff
@@ -549,25 +549,29 @@ enum {
 	 */
 	BT_LE_ADV_OPT_USE_IDENTITY = BIT(2),
 
-	/** Advertise using GAP device name.
-	 *
-	 *  Include the GAP device name automatically when advertising.
-	 *  By default the GAP device name is put at the end of the scan
-	 *  response data.
-	 *  When advertising using @ref BT_LE_ADV_OPT_EXT_ADV and not
-	 *  @ref BT_LE_ADV_OPT_SCANNABLE then it will be put at the end of the
-	 *  advertising data.
-	 *  If the GAP device name does not fit into advertising data it will be
-	 *  converted to a shortened name if possible.
-	 *  @ref BT_LE_ADV_OPT_FORCE_NAME_IN_AD can be used to force the device
-	 *  name to appear in the advertising data of an advert with scan
-	 *  response data.
-	 *
-	 *  The application can set the device name itself by including the
-	 *  following in the advertising data.
-	 *  @code
-	 *  BT_DATA(BT_DATA_NAME_COMPLETE, name, sizeof(name) - 1)
-	 *  @endcode
+	/**
+	 * \deprecated This option will be removed in the near future, see
+	 * https://github.com/zephyrproject-rtos/zephyr/issues/71686
+	 *
+	 * @brief Advertise using GAP device name.
```

Comment:
```suggestion
	/** @brief Advertise using GAP device name.
	 * 
	 *  @deprecated This option will be removed in the near future, see
	 *  https://github.com/zephyrproject-rtos/zephyr/issues/71686
	 *
	 * 
```

I would revert back to this and add deprecated below the @brief to avoid this many line changes.

Reply:
I think it's nice to have it at the top. In the documentation you instantly see that the macro is deprecated.